docs/reference/search/request/search-after.asciidoc

```diff
@@ -0,0 +1,59 @@
+[[search-request-search-after]]
+=== Search After
+
+Pagination of results can be done by using the `from` and `size` parameters but it can be very costly for indices with
+more than one shard if the `from` parameter is big. The `search_after` parameter has been added to address this problem,
+it provides a way to efficiently paginate from one page to the next. This parameter accepts an array of sort values,
+those values are then used by the searcher to sort the top hits from the first document that is greater than the sort values.
+This parameter must be used in conjunction with the sort parameter, it must contain exactly the same number of values
+than the number of fields to sort on.
+
+NOTE: The parameter `from` must be set to 0 (or -1) when `search_after` is used.
+
+NOTE: A field with one unique value per document should be used as the last element of the sort specification.
+Otherwise the sort order for documents that have the same sort values would be undefined. The recommended way is to use
+the field `_uid` which is certain to contain one unique value for each document.
+
+Here is an example of how to use it:
+
+Suppose that the query to retrieve the first page looks like this:
+[source,js]
+--------------------------------------------------
+curl -XGET 'localhost:9200/twitter/tweet/_search'
+{
+    "query": {
+        "match" : {
+            "title" : "elasticsearch"
+        }
+    },
+    "sort": [
+        {"age": "asc"},
+        {"_uid": "desc"}
+    ]
+}
+'
+--------------------------------------------------
+
+The result from the above request includes an array of `sort` values for each document.
+In order to retrieve the next page we can use the `sort` values of the last document and pass it to the next request to
+ retrieve the next page of results:
+[source,js]
+--------------------------------------------------
+curl -XGET 'localhost:9200/twitter/tweet/_search'
+{
+    "query": {
+        "match" : {
+            "title" : "elasticsearch"
+        }
+    },
+    "search_from": [18, "tweet#654323"],
+    "sort": [
+        {"age": "asc"},
+        {"_uid": "desc"}
+    ]
+}
+'
```

Comment:
Could you make a rest test out of this example so that we also test the rest layer with this feature?